### ISSUE TITLE
Update Gradle plugin to 3.3.0

### ISSUE THREAD

**Issue by aforcier** [woocommerce/woocommerce-android#694]
As of this writing, `3.3.0` is the latest gradle plugin. We should update the project to use it. However, at the moment updating generates some build warnings.

The first warning, `WARNING: API 'variant.getGenerateBuildConfig' is obsolete`, can be resolved by using the new provider config for this Crashlytics task:

```groovy
android.applicationVariants.all { variant ->
    variant.generateBuildConfig.dependsOn(generateCrashlyticsConfig)
}
```

to
```groovy
android.applicationVariants.all { variant ->
    variant.generateBuildConfigProvider.configure {
        dependsOn(generateCrashlyticsConfig)
    }
}
```

The next warning requires the Fabric gradle plugin to be updated to support `3.3.0` and fix its internal warnings - we should keep an eye on this changelog: https://docs.fabric.io/android/changelog.html#fabric-gradle-plugin, then update the version we use along with our Gradle version.

**Comment by aforcier** [comment]
Note: The gradle version has already updated to `3.3.0`, but this issue now tracks what we need to do to silence the `obsolete` warnings we're getting when building.

**Comment by aforcier** [comment]
Fixed in #911.
